addons/l10n_ch/tests/test_gen_isr_reference.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import common, Form
+
+QR_IBAN = 'CH21 3080 8001 2345 6782 7'
+ISR_SUBS_NUMBER = "01-162-8"
+
+
+class TestGenISRReference(common.SavepointCase):
+    """Check condition of generation of and content of the structured ref"""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.partner = cls.env.ref("base.res_partner_12")
+        cls.bank = cls.env["res.bank"].create(
+            {
+                "name": "Alternative Bank Schweiz AG",
+                "bic": "ALSWCH21XXX",
+            }
+        )
+        cls.bank_acc_isr = cls.env["res.partner.bank"].create(
+            {
+                "acc_number": "ISR",
+                "l10n_ch_isr_subscription_chf": "01-162-8",
+                "bank_id": cls.bank.id,
+                "partner_id": cls.partner.id,
+            }
+        )
+        cls.bank_acc_qriban = cls.env["res.partner.bank"].create(
+            {
+                "acc_number": QR_IBAN,
+                "bank_id": cls.bank.id,
+                "partner_id": cls.partner.id,
+            }
+        )
+
+    def new_form(self):
+        inv = Form(self.env["account.move"].with_context(
+            default_type="out_invoice")
+        )
+        inv.partner_id = self.partner
+        inv.currency_id = self.env.ref("base.CHF")
+        with inv.invoice_line_ids.new() as line:
+            line.name = "Fondue"
+            line.price_unit = 494.
+        return inv
+
+    def test_isr(self):
+        inv_form = self.new_form()
+        inv_form.invoice_partner_bank_id = self.bank_acc_isr
+        invoice = inv_form.save()
+
+        invoice.name = "INV/01234567890"
+
+        expected_isr = "000000000000000012345678903"
+        expected_optical_line = (
+            "0100000494004>000000000000000012345678903+ 010001628>"
+        )
```

Comment:
Is it just to make the line shorter? Please inline it, it won't harm readability. Those parenthesis are confusing.

Reply:
Not confusing for people used to 80ish line len I guess :smile: 

I'll remove them.